src/shallow/validateShallowOptions.lua

```diff
@@ -0,0 +1,30 @@
+local optionsTypes = {
+	depth = "number",
+}
+
+local function validateShallowOptions(options)
+	if options == nil then
+		return true
+	end
+
+	for key, value in pairs(options) do
+		local expectType = optionsTypes[key]
+
+		if expectType == nil then
+			return false, ("unexpected option field %q (with value of %s)"):format(
+				tostring(key),
+				tostring(value)
+			)
+		elseif typeof(value) ~= expectType then
+			return false, ("unexpected option type for %q (expected %s but got %s)"):format(
+				tostring(key),
+				expectType,
+				typeof(value)
+			)
+		end
+	end
+
+	return true
+end
```

Comment:
This solution seems way overzealous for something like this. Is this in the name of future proofing? I worry it's getting a little out of hand.

Reply:
Yes at some point I add three parameters (depth, hostKey and hostParent) so I felt like it would be good to make sure there are no typos and mismatched types. But yeah right now with only the depth it looks a bit like nuking a fly.

Reply:
Should I still keep the dictionary as a parameter for future proofing, but remove the validation?

Reply:
I think when it comes to stuff like this, we need to strike a balance between predicting how our API might change in the future and just giving up and making it a table. Tables are fine in a lot of cases but they feel really awkward when there are only one or two elements.

Personally, I'm okay with the public `shallow` function just taking depth as its second argument. It doesn't seem like we would wildly change this function's signature anytime soon. What do you think?

Reply:
Yes and you make me realise that since we are now mounting with `Roact.mount`, the shallow function is not public anymore. I use it as en entry point from Roact to the shallow module.